pkg/apis/submariner.io/v1/types.go

```diff
@@ -175,3 +175,79 @@ func (c *Connection) SetStatus(status ConnectionStatus, messageFormat string, a
 	c.Status = status
 	c.StatusMessage = fmt.Sprintf(messageFormat, a...)
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type GlobalnetEgressIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the specification of the desired behavior.
+	Spec GlobalnetEgressIPSpec `json:"spec"`
+
+	// The most recently observed status. Read-only.
+	// +optional
+	Status GlobalnetEgressIPStatus `json:"status,omitempty"`
+}
+
+type GlobalnetEgressIPSpec struct {
+	// The specific GlobalIP to request. This should be a valid IP address from the Globalnet CIDR assigned to the cluster.
+	// If the requested GlobalIP falls outside of the Globalnet CIDR or is already allocated (either fully or partially),
+	// allocation will fail with an appropriate Status condition set.
+	// If not specified, GlobalIP(s) will be auto-allocated according to the NumGlobalIPs field.
+	// +optional
+	GlobalIP string `json:"globalIP,omitempty"`
+
+	// The requested number of contiguous GlobalIPs to allocate from the Globalnet CIDR assigned to the cluster.
+	// This field is ignored if the GlobalIP field is set. Defaults to 1.
+	// +optional
+	NumGlobalIPs int `json:"numGlobalIPs,omitempty"`
+
+	// Selects the specific namespaces to which this GlobalnetEgressIP applies. Pods from matching namespaces
+	// will have GlobalIPs assigned.
+	// If the selector is empty, all namespaces are selected.
+	// If not specified, the namespace in which the GlobalnetEgressIP is deployed is used.
+	// +optional
+	NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty"`
+
+	// Selects specific pods in the namespaces matched by the NamespaceSelector.
+	// If the selector is empty or not specified, all pods are selected.
+	// If a pod matches multiple GlobalnetEgressIP objects, there is no guarantee from which GlobalnetEgressIP its
+	// GlobalIP will be assigned.
+	// +optional
+	PodSelector *metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+
+type GlobalnetEgressIPConditionType string
+
+const (
+	GlobalnetEgressIPAllocated GlobalnetEgressIPConditionType = "Allocated"
+)
+
+type GlobalnetEgressIPStatus struct {
+	// Condition contains details for the current condition of this GlobalnetEgressIP.
+	// +optional
+	//Condition metav1.Condition `json:"condition,omitempty"`
```

Comment:
metav1.Condition will be available once we bump to K8s 0.19.* (https://github.com/submariner-io/shipyard/pull/539).